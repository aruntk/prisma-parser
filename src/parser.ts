import * as pegjs from 'pegjs'
import grammar from './grammar'
import { IPrismaAST } from './interface'
import printer from './printer'

const parser = pegjs.generate(grammar)
export default {
  parse: (prismaQL: string): IPrismaAST => parser.parse(prismaQL),
  print: printer,
}
