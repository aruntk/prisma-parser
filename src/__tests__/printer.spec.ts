import parser from '../parser'

const multipleLinesBetweenScopesSchema = `datasource db1 {
  provider = "mysql"
  url = "mysql://localhost:3306"
}



datasource db2 {
  provider = "postgresql"
  url = "postgresql://localhost:5432"
}`

const multipleLinesBetweenDeclarationsSchema = `datasource db1 {
  provider = "mysql"



  url = "mysql://localhost:3306"
}`

const multipleLinesBetweenColumnsSchema = `model User {
  id Int @id


  name String
}`

const indentationCheckSchema = `
    model User    {
          id     Int   @id
name   String
    }`

const completeSchema = `datasource db1 {
  provider = "mysql"
  url = "mysql://localhost:3306"
}
datasource db2 {
  provider = "postgresql"
  url = "postgresql://localhost:5432"
}
datasource db3 {
  provider = "sqlite"
  url = "file:dev.db"
  enabled = true
}
generator photon {
  provider = "photonjs"
}
model User {
  id     Int @id
  name   String
  email  String
  age    Int?
  posts  Post[]
}
model Post {
  id       Int @id
  title    String
  content  String
  author   User
}`

describe('PrismaPrinter', () => {
  it('Prints multiple new lines between scopes correctly', async () => {
    const tree = parser.parse(multipleLinesBetweenScopesSchema)
    const printerOut = parser.print(tree)
    expect(printerOut).toMatchSnapshot()
  })

  it('Prints multiple new lines between declarations correctly', async () => {
    const tree = parser.parse(multipleLinesBetweenDeclarationsSchema)
    const printerOut = parser.print(tree)
    expect(printerOut).toMatchSnapshot()
  })

  it('Prints multiple new lines between columns correctly', async () => {
    const tree = parser.parse(multipleLinesBetweenColumnsSchema)
    const printerOut = parser.print(tree)
    expect(printerOut).toMatchSnapshot()
  })

  it('Prints indentation correctly', async () => {
    const tree = parser.parse(indentationCheckSchema)
    const printerOut = parser.print(tree)
    expect(printerOut).toMatchSnapshot()
  })

  it('Prints ast -> schema correctly for a complete example', async () => {
    const tree = parser.parse(completeSchema)
    const printerOut = parser.print(tree)
    expect(printerOut).toMatchSnapshot()
  })
})
