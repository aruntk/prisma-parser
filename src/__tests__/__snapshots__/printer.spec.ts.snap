// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrismaPrinter Prints ast -> schema correctly 1`] = `
"datasource db1 {
  provider = \\"mysql\\"
  url = \\"mysql://localhost:3306\\"
}
datasource db2 {
  provider = \\"postgresql\\"
  url = \\"postgresql://localhost:5432\\"
}
datasource db3 {
  provider = \\"sqlite\\"
  url = \\"file:dev.db\\"
  enabled = true
}
generator photon {
  provider = \\"photonjs\\"
}
model User {
  id Int @id
  name String
  email String
  age Int?
  posts Post[]
}
model Post {
  id Int @id
  title String
  content String
  author User
}"
`;
